crates/oxc_linter/src/rules/import/namespace.rs

```diff
@@ -1,30 +1,104 @@
+    #[diagnostic(severity(warning))]
+    ComputedReference(#[label] Span, String),
+    #[error("eslint-plugin-import(namespace): Assignment to member of namespace {1:?}.'")]
+    #[diagnostic(severity(warning))]
+    Assignment(#[label] Span, String),
+}
 
 /// <https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/namespace.md>
 #[derive(Debug, Default, Clone)]
 pub struct Namespace;
 
 declare_oxc_lint!(
     /// ### What it does
-    /// TODO
+    /// Enforces names exist at the time they are dereferenced, when imported as a full namespace (i.e. import * as foo from './foo'; foo.bar(); will report if bar is not exported by ./foo.).
+    /// Will report at the import declaration if there are no exported names found.
+    /// Also, will report for computed references (i.e. foo["bar"]()).
+    /// Reports on assignment to a member of an imported namespace.
     Namespace,
     nursery
 );
 
 impl Rule for Namespace {
-    fn run_once(&self, _ctx: &LintContext<'_>) {}
+    fn run(&self, node: &AstNode<'_>, ctx: &LintContext<'_>) {
+        if let AstKind::ImportNamespaceSpecifier(specifier) = node.kind() {
+            let Some(AstKind::ImportDeclaration(decl)) = ctx.nodes().parent_kind(node.id()) else {
+                return;
+            };
+
+            let source = &decl.source.value;
+            let module_record = ctx.semantic().module_record();
+            let Some(module) = module_record.loaded_modules.get(source.as_ref()) else {
+                return;
+            };
+
+            let check_binding_exported = |name: &str, span| {
+                if module.exported_bindings.get(name).is_none() {
+                    ctx.diagnostic(NamespaceDiagnostic::NoExport(
+                        span,
+                        name.to_owned(),
+                        source.to_string(),
+                    ));
+                }
+            };
+
+            if let Some(symbol_id) = specifier.local.symbol_id.get() {
+                ctx.symbols().get_resolved_references(symbol_id).for_each(|reference| {
+                    if let Some(node) = ctx.nodes().parent_node(reference.node_id()) {
+                        let name = specifier.local.name.clone().to_string();
+
+                        match node.kind() {
+                            AstKind::MemberExpression(member) => {
+                                if matches!(
+                                    ctx.nodes().parent_kind(node.id()),
+                                    Some(AstKind::SimpleAssignmentTarget(_))
+                                ) {
+                                    ctx.diagnostic(NamespaceDiagnostic::Assignment(
+                                        member.span(),
+                                        name.clone(),
+                                    ));
+                                };
+
+                                if member.is_computed() {
```

Comment:
This is behind a configuration https://github.com/import-js/eslint-plugin-import/blob/ee5fadeffff68f2300bed7f67a310496cb969d61/src/rules/namespace.js#L127

the default is false, can you add a TODO somewhere?